src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -154,17 +154,23 @@ def _prompt_for_parameters(missing_parameters):
     return {}
 
 def _deploy_arm_template_core(resource_group_name, template_file=None, template_uri=None,
-                              deployment_name=None, parameters=None, mode='incremental',
+                              deployment_name=None, parameter_list=None, mode='incremental',
                               validate_only=False, no_wait=False):
     from azure.mgmt.resource.resources.models import DeploymentProperties, TemplateLink
 
     if bool(template_uri) == bool(template_file):
         raise CLIError('please provide either template file path or uri, but not both')
 
-    if parameters:
-        parameters = json.loads(parameters)
-        if parameters:
-            parameters = parameters.get('parameters', parameters)
+    parameters = None
+    if parameter_list:
+        for params in parameter_list:
+            params_object = json.loads(params)
+            if params_object:
+                params_object = params_object.get('parameters', params_object)
+            if parameters is None:
+                parameters = params_object
+            else:
+                parameters.update(params_object)
```

Comment:
Added this to the description of the `--parameters` flag. Added some tests. Let me know if you want anything more here.